cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1353,6 +1353,8 @@ func getNonUEFISystemDisk(fallbacklabel string) (string, error) {
 		}
 		return partition.KernelDeviceNode, nil
 	}
+	// TODO Read files in folder and check for case-insensitive matches, for vfat
+	// (maybe unnecessary as this is a legacy code path?)
```

Comment:
there is code based on by-label also in gadget/device_linux.go

I'm also not sure we can define this the legacy path, don't think many things are yet using snapd_system_disk in the field

Reply:
I looked at the code in `device_linux.go` and at first sight it seemed ok, as it is matching `VolumeStructure.Label`, which is read from the gadget, with what is in by-label - they should match as the first is used by ubuntu-image to create the filesystem. I'll check what can we do in initramfs.

Reply:
Changed, looking now at all possible valid labels

Reply:
FindDeviceForStructure seems to be used by update so it might involve also things not created by snapd/ubuntu-image, so I think that code needs changes too? we probably need a gadget update test for a seed asset when seed fslabel is uppercase?